WordPress/src/main/java/org/wordpress/android/ui/bloggingreminders/BloggingRemindersAdapter.kt

```diff
@@ -3,18 +3,37 @@ package org.wordpress.android.ui.bloggingreminders
 import android.view.ViewGroup
 import androidx.recyclerview.widget.DiffUtil
 import androidx.recyclerview.widget.RecyclerView.Adapter
+import org.wordpress.android.ui.bloggingreminders.BloggingRemindersItem.CloseButton
+import org.wordpress.android.ui.bloggingreminders.BloggingRemindersItem.Illustration
+import org.wordpress.android.ui.bloggingreminders.BloggingRemindersItem.PrimaryButton
+import org.wordpress.android.ui.bloggingreminders.BloggingRemindersItem.Text
+import org.wordpress.android.ui.bloggingreminders.BloggingRemindersItem.Title
 import org.wordpress.android.ui.bloggingreminders.BloggingRemindersItem.Type
+import org.wordpress.android.ui.bloggingreminders.BloggingRemindersItem.Type.CLOSE_BUTTON
+import org.wordpress.android.ui.bloggingreminders.BloggingRemindersItem.Type.ILLUSTRATION
+import org.wordpress.android.ui.bloggingreminders.BloggingRemindersItem.Type.PRIMARY_BUTTON
+import org.wordpress.android.ui.bloggingreminders.BloggingRemindersItem.Type.TEXT
 import org.wordpress.android.ui.bloggingreminders.BloggingRemindersItem.Type.TITLE
+import org.wordpress.android.ui.bloggingreminders.BloggingRemindersViewHolder.CloseButtonViewHolder
+import org.wordpress.android.ui.bloggingreminders.BloggingRemindersViewHolder.IllustrationViewHolder
+import org.wordpress.android.ui.bloggingreminders.BloggingRemindersViewHolder.PrimaryButtonViewHolder
+import org.wordpress.android.ui.bloggingreminders.BloggingRemindersViewHolder.TextViewHolder
+import org.wordpress.android.ui.bloggingreminders.BloggingRemindersViewHolder.TitleViewHolder
+import org.wordpress.android.ui.utils.UiHelpers
+import org.wordpress.android.util.image.ImageManager
+import javax.inject.Inject
 
-class BloggingRemindersAdapter : Adapter<BloggingRemindersViewHolder<*>>() {
+class BloggingRemindersAdapter
```

Comment:
Just curious!  Why is this modeled as an Adapter with ViewHoders?  
Is it is like airbnb [epoxy](https://github.com/airbnb/epoxy)
There's a similar [ConcatAdapter](https://medium.com/androiddevelopers/merge-adapters-sequentially-with-mergeadapter-294d2942127a) in the works but still in Alpha I guess.

Reply:
I tend to model all the UI as a list. It makes it much easier to unit test the UI state, makes building a UI much more flexible and removes a lot of the logic from the Views. I've never used epoxy. Also from a quick look at ConcatAdapter I don't see any benefit over multiple view holders. Are there any?

Reply:
Not really about advantage.  I found your approach similar to epoxy.  I've used epoxy in cases where content comes from backend and screens are built dynamically.  Wondering, if that was the reason why you chose to go with this approach.

ConcatAdapter is similar to airbnb epoxy.  ConcatAdapter allows to combine multiple types of adapters with their own data sources and also reuse viewholders across adapters too. 

Of course, all that might become obsolete when we start using Jetpack Compose components!